examples/e2e/steps/training/model_evaluator.py

```diff
@@ -0,0 +1,113 @@
+    and validation data set pair which it uses to evaluate the model's
+    performance. The model metrics are then returned as step output artifacts
+    (in this case, the model accuracy on the train and test set).
+
+    The suggested step implementation also outputs some warnings if the model
+    performance does not meet some minimum criteria. This is just an example of
+    how you can use steps to monitor your model performance and alert you if
+    something goes wrong. As an alternative, you can raise an exception in the
+    step to force the pipeline run to fail early and all subsequent steps to
+    be skipped.
+
+    This step is parameterized to configure the step independently of the step code,
+    before running it in a pipeline. In this example, the step can be configured
+    to use different values for the acceptable model performance thresholds and
+    to control whether the pipeline run should fail if the model performance
+    does not meet the minimum criteria. See the documentation for more
+    information:
+
+        https://docs.zenml.io/user-guide/advanced-guide/configure-steps-pipelines
+
+    Args:
+        model: The pre-trained model artifact.
+        dataset_trn: The train dataset.
+        dataset_tst: The test dataset.
+        min_train_accuracy: Minimal acceptable training accuracy value.
+        min_test_accuracy: Minimal acceptable testing accuracy value.
+        fail_on_accuracy_quality_gates: If `True` a `RuntimeException` is raised
+            upon not meeting one of the minimal accuracy thresholds.
+
+    Raises:
+        RuntimeError: if any of accuracies is lower than respective threshold
+
+    Returns:
+        The model accuracy on the train and test set.
+    """
+    ### ADD YOUR OWN CODE HERE - THIS IS JUST AN EXAMPLE ###
+    # Calculate the model accuracy on the train and test set
+    trn_acc = model.score(
+        dataset_trn.drop(columns=[MetaConfig.target_column]),
+        dataset_trn[MetaConfig.target_column],
+    )
+    logger.info(f"Train accuracy: {trn_acc}")
+    tst_acc = model.score(
+        dataset_tst.drop(columns=[MetaConfig.target_column]),
+        dataset_tst[MetaConfig.target_column],
+    )
+    logger.info(f"Test accuracy: {tst_acc}")
+
+    messages = []
+    if trn_acc < min_train_accuracy:
+        messages.append(
+            f"Train accuracy {trn_acc*100:.2f}% is below {min_train_accuracy*100:.2f}% !"
+        )
+    if tst_acc < min_test_accuracy:
+        messages.append(
+            f"Test accuracy {tst_acc*100:.2f}% is below {min_test_accuracy*100:.2f}% !"
+        )
+    if fail_on_accuracy_quality_gates and messages:
+        raise RuntimeError(
+            "Model performance did not meet the minimum criteria:\n"
+            + "\n".join(messages)
+        )
+    else:
+        for message in messages:
+            logger.warning(message)
+
+    mlflow.log_metric("testing_accuracy_score", tst_acc)
```

Comment:
You might want to move this up, otherwise the metric is not logged if it's too low

Reply:
Thanks, moved up